frontend/dockerfile/builder/build.go

```diff
@@ -834,8 +834,26 @@ func contextByName(ctx context.Context, c client.Client, name string, platform *
 		if platform != nil {
 			imgOpt = append(imgOpt, llb.Platform(*platform))
 		}
+
+		named, err := reference.ParseNormalizedNamed(ref)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		_, data, err := c.ResolveImageConfig(ctx, named.String(), llb.ResolveImageConfigOpt{
```

Comment:
This should do two config resolves now (they might be cached internally) because llb also uses `llb.WithMetaResolver()`. You should remove that and call https://pkg.go.dev/github.com/moby/buildkit@v0.10.2/client/llb#State.WithImageConfig instead on the state.

Reply:
Ah, I think I understand. Updated, PTAL.